src/operator/matrix_op-inl.h

```diff
@@ -643,6 +654,10 @@ void Slice(const TBlob &src,
   mshadow::Stream<xpu> *s = ctx.get_stream<xpu>();
   int ndim = static_cast<int>(ret->shape_.ndim());
 
+  int axis_size = static_cast<int>(src.shape_[param.axis]);
+  param.begin = (param.begin < 0) ? (param.begin + axis_size + 1) : param.begin;
```

Comment:
Actually: `end=None` is not bad. Its similar to numpy slice: `a[1:]` is equivalent to `being=1, end=None`, as no index is specified in Python. Then we should also have `a[:3]` equivalent to `begin=None, end=3`.